api/src/main/java/jakarta/servlet/http/Cookie.java

```diff
@@ -422,4 +423,101 @@ public void setHttpOnly(boolean isHttpOnly) {
     public boolean isHttpOnly() {
         return isHttpOnly;
     }
+
+    /**
+     * Sets the <i>SameSite</i> attribute for this cookie with the provided value.
+     *
+     * <p>
+     * When <code>sameSite</code> is not <code>null</code>, this cookie will have its <code>SameSite</code>
+     * attribute set with the provided value.
+     *
+     * @param sameSite the <i>SameSite</i> attribute's value; may be null if the attribute should not be set
+     *
+     * @since Servlet 5.0
+     */
+    public void setSameSite(SameSite sameSite) {
+        this.sameSite = sameSite;
+    }
+
+    /**
+     * Returns the <i>SameSite</i> attribute associated with this cookie.
+     *
+     * @return the <i>SameSite</i> attribute associated with this cookie; may be null to indicate that the attribute
+     *         should not be set
+     *
+     * @since Servlet 5.0
+     */
+    public SameSite getSameSite() {
+        return sameSite;
+    }
+
+    /**
+     * The available values that can be used with the cookie's <i>SameSite</i> attribute.
+     */
+    public enum SameSite {
```

Comment:
I think we need to add another value called "DEFAULT" or "CONTAINER".  The meaning of which is that the cookie will use a container provided default for all Cookies set.  This will allow existing apps to have this attribute set for them.

Reply:
This could work, however the literal representation for such value would not make too much sense. Should I add SESSION_COOKIE_DEFAULT/DEFAULT/CONTAINER/CONTAINER_DEFAULT to the mix and remove the literal representation at the same time?

Reply:
Given that the enum will probably need to be used by both the container and the cookie itself, perhaps we don't need to have an extra item, but only need to define what null means.
Currently a null value in `Cookie` means don't have the attribute.  However, if we say that null means - use the containers (or contexts?)  default value for the attribute, then that should work.  The container/context default itself shoould be null (for now) and that should mean no same-site attributed.   But the container/context should be able to be configured to have a none null default and thus change all cookies created by the application.

Putting the default in the "container" makes this a javadoc only feature.  Putting it into the "context" would mean that it would have real API in `ServletContext` and configuration in web.xml and perhaps annotations.   I'm a bit torn between the simplicity of a javadoc only feature vs the large impact of putting it in the context (and then one could ask why don't we do the same for httpOnly and secure cookie attributes?)

thoughts?

Reply:
Exactly what scope is implied by putting it in the container?
Does that mean it's scoped to all applications running in a given Servlet container instance?

Would there be any need to configure it differently for different applications running
in the same server instance?

Would there be any need to configure it differently for different Servlets in a given
application?

Reply:
I assume the need to impose a Same-Site cookie configuration depends on the set of clients that are involved.  Typically that is a function of the container configuration and that any container exposed to clients that will not tolerate an absence of the attribute will have all contexts and servlets on them affected.   There are of course exceptions to that, in that it is always possible for a container to expose some contexts to only a subset of connections, and for a context to expose a servlet to only a subset of request sources.

But in general I would see that a container exposed to clients that need Same-Site would need to add that attribute to all cookies generated by all contexts and all servlets.

Thus my first thought that this should be a container wide default.... but that is javadoc only.
Next option is to expose that container default in a ServletContext getter (and nother stopping a container lieing and giving different values to different contexts), so this becomes visible in the API.
The ultimate step is to make this settable in ServletContext, so an application can change its own default (or that of the container).  I'm not sure this is necessary as it can already change the individual Cookies and session cookie config, so it has the power without a ServletContext setter.

Reply:
IMHO this should be a javadoc only feature. I don't see any benefit to including this in the ServletContext.